### PR TITLE
ekf2: update to latest ecl version

### PR DESCRIPTION
**Pull request by priseborough**
Adds ability to use velocity data from vehicle_odometry message and miscellaneous fixes for use of other data from vehicle_odometry. See https://github.com/PX4/ecl/pull/642 for details

Includes ekf2_main.cpp compatibility changes for renamed variables.

@kamilritz FYI

### CODE REVIEW

**Comment by mhkabir** [comment]
@priseborough This also need to include the copying of velocity data into the ECL backend.

**Comment by priseborough** [comment]
@kamilritz Can you please have a look at the parameter definitions before tomorrow if you get time. The build is failing at the step where the parameter xml file is generated and i am out of time today. TXS.

File: src/modules/ekf2/ekf2_params.c [hunk 1]
```diff
@@ -1161,6 +1157,26 @@ PARAM_DEFINE_FLOAT(EKF2_RNG_A_HMAX, 5.0f);
  */
 PARAM_DEFINE_FLOAT(EKF2_RNG_A_IGATE, 1.0f);
 
+/**
+ * Gate size for vision velocity estimate fusion
+ * Sets the number of standard deviations used by the innovation consistency test.
```
Comment (kamilritz) [inline src/modules/ekf2/ekf2_params.c]: `
* Gate size for vision velocity estimate fusion
*
 * Sets the number of standard deviations used by the innovation consistency test.
 *
  * @group EKF2
`
add some spaces here

File: src/modules/ekf2/ekf2_params.c [hunk 2]
```diff
@@ -1161,6 +1157,26 @@ PARAM_DEFINE_FLOAT(EKF2_RNG_A_HMAX, 5.0f);
...
+ * @group EKF2
+ * @min 1.0
+ * @Unit SD
```
Comment (kamilritz) [inline src/modules/ekf2/ekf2_params.c]:
```suggestion
 * @unit SD
```

File: src/modules/ekf2/ekf2_params.c [hunk 3]
```diff
@@ -1161,6 +1157,26 @@ PARAM_DEFINE_FLOAT(EKF2_RNG_A_HMAX, 5.0f);
...
+ * @decimal 1
+*/
+PARAM_DEFINE_FLOAT(EKF2_EVV_GATE, 3.0f);
+
+/**
+ * Gate size for vision position fusion
```
Comment (kamilritz) [inline src/modules/ekf2/ekf2_params.c]: Add some empty lines here as well

**Comment by jkflying** [comment]
continued in #13023

**Pull request closed without merging**